test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/IdentityHashMapTest.scala

```diff
@@ -0,0 +1,397 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.javalib.util
+
+import java.{util => ju}
+import scala.reflect.ClassTag
+import org.junit.Assert._
+import org.junit.Test
+
+class IdentityHashMapTest extends MapTest {
+  import MapTest._
+
+  override def factory: IdentityHashMapFactory = new IdentityHashMapFactory
+
+  private val testObjCache = new ju.HashMap[Int, TestObj]()
+
+  override def testObj(i: Int): TestObj = {
+    val existing = testObjCache.get(i)
+    if (existing ne null) {
+      existing
+    } else {
+      val newTestObj = TestObj(i)
+      testObjCache.put(i, newTestObj)
+      newTestObj
+    }
+  }
+
+  def assumeFalseIdentityHashMapViews(): Unit = assumeFalseIdentityHashMap("JVM views removeAll, retainAll")
+
+  // Needs false negative tests
+  // Tests that check by value rather than identity
+  // Similar to tests already in MapTest
+
+  @Test def testIdentityEntrySetIsViewForQueriesWithCustomObjects(): Unit = {
+    val mp = factory.empty[TestObj, TestObj]
+    mp.put(testObj(1), testObj(11))
+    mp.put(testObj(2), testObj(22))
+    mp.put(testObj(5), null)
+    mp.put(null, testObj(55))
+    val entrySet = mp.entrySet()
+
+    assertTrue(entrySet.contains(SIE(testObj(1), testObj(11))))
+    assertFalse(entrySet.contains(SIE(testObj(1), TestObj(11))))
+    assertFalse(entrySet.contains(SIE(TestObj(1), testObj(11))))
+
+    assertTrue(entrySet.contains(SIE(testObj(5), null)))
+    assertFalse(entrySet.contains(SIE(TestObj(5), null)))
+    assertFalse(entrySet.contains(SIE(testObj(6), null)))
+
+    assertTrue(entrySet.contains(SIE(null, testObj(55))))
+    assertFalse(entrySet.contains(SIE(null, TestObj(55))))
+
+    mp.put(testObj(3), testObj(33))
+
+    // containsAll calls contains tested above
+    val coll1 = ju.Arrays.asList(SIE(testObj(1), testObj(11)),
```

Comment:
Consider using the new `TrivialImmutableCollection` instead of `ju.Arrays.asList`. (This applies to all uses of `ju.Arrays.asList` in this file.)

Reply:
Definitely cleaner, thanks.